enclave-runtime/src/lib.rs

```diff
@@ -383,57 +385,164 @@ pub unsafe extern "C" fn get_state(
 		connection_registry.clone(),
 		state_handler.clone(),
 		ocall_api.clone(),
-		rsa_shielding_key,
+		shielding_key,
 	);
-
 	GLOBAL_RPC_AUTHOR_COMPONENT.initialize(rpc_author.clone());
 
-	let stf_executor = Arc::new(EnclaveStfExecutor::new(ocall_api, state_handler));
 	let top_pool_operation_handler =
-		Arc::new(EnclaveTopPoolOperationHandler::new(rpc_author.clone(), stf_executor));
-
+		Arc::new(EnclaveTopPoolOperationHandler::new(rpc_author.clone(), stf_executor.clone()));
 	GLOBAL_TOP_POOL_OPERATION_HANDLER_COMPONENT.initialize(top_pool_operation_handler);
 
-	let io_handler = public_api_rpc_handler(rpc_author);
+	let io_handler = public_api_rpc_handler(rpc_author.clone());
 	let rpc_handler = Arc::new(RpcWsHandler::new(io_handler, watch_extractor, connection_registry));
+	GLOBAL_RPC_WS_HANDLER_COMPONENT.initialize(rpc_handler);
+
+	let top_pool_executor = Arc::<EnclaveTopPoolOperationHandler>::new(
+		TopPoolOperationHandler::new(rpc_author.clone(), stf_executor),
+	);
+
+	let parentchain_block_import_dispatcher = match GLOBAL_PARENTCHAIN_IMPORT_DISPATCHER_COMPONENT
+		.get()
+	{
+		Some(a) => a,
+		None => {
+			error!("Failed to retrieve global parentchain import dispatcher component (maybe it is not initialized?)");
+			return sgx_status_t::SGX_ERROR_UNEXPECTED
+		},
+	};
+
+	let sidechain_block_importer = Arc::<EnclaveSidechainBlockImporter>::new(BlockImporter::new(
+		state_handler,
+		state_key,
+		signer.clone(),
+		top_pool_executor,
+		parentchain_block_import_dispatcher,
+		ocall_api.clone(),
+	));
+
+	let sidechain_block_syncer =
+		Arc::new(EnclaveSidechainBlockSyncer::new(sidechain_block_importer, ocall_api));
+	GLOBAL_SIDECHAIN_BLOCK_SYNCER_COMPONENT.initialize(sidechain_block_syncer.clone());
+
+	let sidechain_block_import_queue = Arc::new(EnclaveSidechainBlockImportQueue::default());
+	GLOBAL_SIDECHAIN_IMPORT_QUEUE_COMPONENT.initialize(sidechain_block_import_queue.clone());
+
+	let sidechain_block_import_queue_worker =
+		Arc::new(EnclaveSidechainBlockImportQueueWorker::new(
+			sidechain_block_import_queue,
+			sidechain_block_syncer,
+		));
+	GLOBAL_SIDECHAIN_IMPORT_QUEUE_WORKER_COMPONENT.initialize(sidechain_block_import_queue_worker);
+
+	let block_composer = Arc::new(BlockComposer::new(signer, state_key, rpc_author));
+	GLOBAL_SIDECHAIN_BLOCK_COMPOSER_COMPONENT.initialize(block_composer);
+
+	sgx_status_t::SGX_SUCCESS
+}
+
+/// Call this once at worker startup to initialize the TOP pool and direct invocation RPC server.
+///
+/// This function will run the RPC server on the same thread as it is called and will loop there.
+/// That means that this function will not return as long as the RPC server is running. The calling
+/// code should therefore spawn a new thread when calling this function.
+#[no_mangle]
+pub unsafe extern "C" fn init_direct_invocation_server(
+	server_addr: *const u8,
+	server_addr_size: usize,
+) -> sgx_status_t {
```

Comment:
This function now only initializes the direct invocation server, as its name suggests. All the component initialization, that used to be here, is moved to the separate functions.